### PR TITLE
[MNT] Switch test configs on MATRIXDESIGN

### PR DESCRIPTION
**Pull request by TonyBagnall**
Reopened this due to overnight testing. The idea is to switch on MATRIXDESIGN. If True, we try minimal configurations aimed at coverage rather than completeness. If False, it does the full tests. We then run the full tests with configs on the overnight runs, and a much faster daily test. 

test_all_forecasters:
uses the forecasting/tests/_config.py for the the matrix tests, forecasting/tests/_config_full.py for the overnight tests. 
 
#59 #151 #154 #271

to come: running off main test times 
test_all_forecasters:
test_all_estimators: 
test_all_classifiers:

running with MATRIXDESIGN:
test_all_forecasters:
test_all_estimators: 
test_all_classifiers:

### CODE REVIEW

**Review by ltsaprounis — COMMENTED** [review]
Left some suggestions mainly for adding some things back in the config or replacing something with something else. This is because so of the removed test_configs are things that can fail if someone adds a new forecaster etc.

**Comment by TonyBagnall** [comment]
> 

great, thanks

File: aeon/forecasting/tests/_config.py [hunk 1]
```diff
@@ -67,43 +67,35 @@
...
     # index type, fh type, is_relative
     ("int", "int", True),
     ("int", "int", False),
-    ("range", "int", True),
-    ("range", "int", False),
-    ("period", "int", True),
-    ("period", "period", False),
-    ("datetime", "int", True),
-    ("datetime", "datetime", False),
-    ("datetime", "timedelta", True),
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/forecasting/tests/_config.py:106]: I guessed as much, why I commented it rather than delete :)

**Comment by TonyBagnall** [comment]
@ltsaprounis I think I reverted those, could you check when you get a chance?

**Comment by TonyBagnall** [comment]
these reverts mean we are back to 17k tests and taking an hour. It has at least identified the prime suspect, testing nine variants of VALID_INDEX_FH_COMBINATIONS creates 10k extra tests and takes it back to an hour. If I run with just these

VALID_INDEX_FH_COMBINATIONS = [
    # index type, fh type, is_relative
    ("int", "int", True),
    ("int", "int", False),
 #   ("range", "int", True),
 #   ("range", "int", False),
 #   ("period", "int", True),
 #   ("period", "period", False),
 #   ("datetime", "int", True),
 #   ("datetime", "datetime", False),
 #   ("datetime", "timedelta", True),
]

it takes 15 mins. I will close this PR for now, I feel there is no enthusiasm for the _confg and_config_full solution, it is a hack really, and the important BATS param changes are now in https://github.com/aeon-toolkit/aeon/pull/568, which does not change coverage. I think we should use the above settings but introduce a one off test that just checks 
VALID_INDEX_FH_COMBINATIONS
without all combinations.

File: aeon/forecasting/tests/_config.py [hunk 2]
```diff
@@ -87,29 +85,19 @@
...
     ("int", "int", True),
-    ("int", "int", False),
-    ("range", "int", True),
-    ("range", "int", False),
-    ("period", "int", True),
-    ("period", "period", False),
-    ("datetime", "int", True),
     ("datetime", "datetime", False),
-    ("datetime", "timedelta", True),
 ]
```
Comment (MatthewMiddlehurst) [inline aeon/forecasting/tests/_config.py:98]: based on the above comments the main issue seems to be this, may be worth adding back some of the important parameters based on @ltsaprounis comments above?

**Review by MatthewMiddlehurst — COMMENTED** [review]
I think this is good in principle, but am unsure about the actual values for each config set.

Ideally, we want the most common/easy/important failures in the PR config

File: aeon/forecasting/tests/_config.py (hunk 2 above)
Comment (TonyBagnall) [reply aeon/forecasting/tests/_config.py:98]: I have no feel for it tbh, happy for anyone to advise, but remember this is meant to be minimal combinations and the expectation is people run test_all_forecasters before a forecasting PR

File: aeon/forecasting/tests/_config_full.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    "TEST_INS_FHS",
+    "TEST_OOS_FHS",
+    "TEST_WINDOW_LENGTHS_INT",
+    "TEST_WINDOW_LENGTHS",
+    "TEST_INITIAL_WINDOW_INT",
+    "TEST_INITIAL_WINDOW",
+    "VALID_INDEX_FH_COMBINATIONS",
+    "INDEX_TYPE_LOOKUP",
+    "TEST_N_ITERS",
+]
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/forecasting/tests/_config_full.py:21]: good spot, was there from ages ago I suspect

File: aeon/forecasting/tests/_config.py [hunk 4]
```diff
@@ -2,52 +2,35 @@
...
-    pd.to_datetime(["2000-01-04", "2000-01-08", "2000-01-11"]),
+    pd.to_datetime(["2000-01-23"]),
+    pd.to_datetime(["2000-01-04", "2000-01-08"]),
 ]
 TEST_CUTOFFS = [*TEST_CUTOFFS_INT_LIST, *TEST_CUTOFFS_INT_ARR, *TEST_CUTOFFS_TIMESTAMP]
 
-TEST_WINDOW_LENGTHS_INT = [1, 5]
-TEST_WINDOW_LENGTHS_TIMEDELTA = [pd.Timedelta(1, unit="D"), pd.Timedelta(5, unit="D")]
-TEST_WINDOW_LENGTHS_DATEOFFSET = [pd.offsets.Day(1), pd.offsets.Day(5)]
+TEST_WINDOW_LENGTHS_INT = [1]
```
Comment (MatthewMiddlehurst) [inline aeon/forecasting/tests/_config.py:22]: Maybe this should still be 5 to test a different length?

**Review by MatthewMiddlehurst — COMMENTED** [review]
Think this is fine for now, one comment.

**Review by MatthewMiddlehurst — APPROVED** [review]

File: aeon/forecasting/tests/_config.py (hunk 4 above)
Comment (TonyBagnall) [reply aeon/forecasting/tests/_config.py]: changed back to [1 5[

**Pull request merged**